index.js

```diff
@@ -154,10 +154,22 @@ function packAndInstallLibrary(name, dir, targetDir) {
     fs.mkdirSync(libDestDir, { recursive: true })
 
     const tmpName = name.replace(/[\s\/]/g, "-").replace(/@/g, "")
+
+   
+
     // npm replaces @... with at- where yarn just removes it, so we test for both files here
     const regex = new RegExp(`^(at-)?${tmpName}(.*).tgz$`)
 
-    const packagedName = fs.readdirSync(dir).find(file => regex.test(file))
+    let packagedName = fs.readdirSync(dir).find(file => regex.test(file))
+
+    if(hasYarn) {
+      let yarnVersion = spawn.sync(['-v'], { cwd: dir, encoding: "utf8" });
+
+      if(yarnVersion.stdout.replace('\n','').split('.')[0] >= 3) {
```

Comment:
Non strict comparison with type coercion is a strongly discouraged because it is very error prone. Can we do it in a more safe manner?

Reply:
Yeah, I can use a semver parsing library if that works. I'll work on this over the next couple of days.